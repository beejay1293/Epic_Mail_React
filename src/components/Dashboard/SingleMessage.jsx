import React from 'react';

import propTypes from 'prop-types';

const SingleMessage = ({
  style, name, messages, time, click, click2, del
}) => (

  <li className={style} >
    <input type="checkbox" className="checkbox" />
    <h1 className="name" onClick={click}>{name}</h1>
    <h1 className="message" onClick={click}>{messages}</h1>
    <h1 className="time" onClick={click}>{time}</h1>
    <div className="delete__icon" onClick={del}>
      <i className="fas fa-trash-alt" />
    </div>
  </li>
);

SingleMessage.propTypes = {
  style: propTypes.string.isRequired,
  name: propTypes.string.isRequired,
  messages: propTypes.string.isRequired,
  time: propTypes.string.isRequired,
};
export default SingleMessage;
