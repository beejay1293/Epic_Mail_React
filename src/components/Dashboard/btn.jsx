import React, { PureComponent } from 'react';
import SingleBtn from './SingleBtn';
import { changeDashboardState } from '../../actions/index'
import { connect } from 'react-redux';

export class Btn extends PureComponent {

    state = {
      bts: [
        {
          className: '',
          fonts: 'fas fa-inbox',
          name: 'Inbox',
          id: 1,

        },
        {
          className: '',
          fonts: 'fas fa-paper-plane',
          name: 'sent',
          id: 2,
        },
        {
          className: '',
          fonts: 'fas fa-comment-alt',
          name: 'draft',
          id: 3,
        },
        {
          className: '',
          fonts: 'fas fa-envelope',
          name: 'unread',
          id: 4,
        },
      ],
    };
 
 
  
 
  render() {
    const { bts } = this.state;
    return (
      <div className="nav2">
        <ul>
          {bts.map(btn => (
            <SingleBtn
              styles={this.props.Dashboardstate === btn.name ? 'active' : ''}
              font={btn.fonts}
              name={btn.name}
              key={btn.id}
              click={(type) => this.props.changeState(btn.name)}
             />
          ))}
          <hr />
          <SingleBtn
            styles="contacts__btn"
            font="fas fa-address-book"
            name="Contacts(5)"
           />
          <hr />

          <div className="grp">
            <li className="groups">
              <i className="fas fa-angle-up  up" />
              <h4 className="group__wrapper">Groups</h4>
            </li>
            <ul className="all__groups" />

            <li className="create__group">
              <i className="fas fa-plus" />
Create Group
            </li>
            <hr />
          </div>


        </ul>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  isAuthenticated: state.messages.isAuthenticated,
  inbox: state.messages.inbox,
  sent: state.messages.sent,
  unread: state.messages.unread,
  Dashboardstate: state.messages.dashboardstate,
  errors: state.messages.errors
})

const mapDispatchToProps = {
    changeState: (type) => changeDashboardState(type),
}

export default connect(mapStateToProps, mapDispatchToProps)(Btn);
