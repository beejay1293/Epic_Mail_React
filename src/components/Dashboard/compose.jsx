import React from 'react';

const Compose = ({click}) => (
  <div className="compose__message_wrapper">
    <div className="compose__message_btn" onClick={click}>
            COMPOSE
      <div className="plus__icon"><i className="fas fa-plus" /></div>
    </div>
  </div>
);

export default Compose;
