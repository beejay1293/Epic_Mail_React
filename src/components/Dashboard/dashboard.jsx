import React, { Component } from 'react';
import { css } from '@emotion/core';
import { Redirect } from 'react-router-dom';
import Modal from 'react-modal';
import ScaleLoader from 'react-spinners/ScaleLoader';
import Header from './header';
import SideBar from './sidebar';
import Main from './mainbody';
import { messages, sendMessageAction } from '../../actions';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';

export class Dashboard extends Component {
  state = {
    modal: 'close',
    to: '',
    subject:'',
    message: '',
    success: '',
    error: '',
    sidebarstate: true
  };

 

  handleModal(type) {
    if(type === 'close'){
      this.setState({
        to: '',
        subject:'',
        message: '',
        success: '',
        error: ''
      })
    }
    this.setState({modal: type, sidebarstate: true})
    
  }

  handleInput(e) {
    const { name, value } = e.target
    this.setState({[name]: value})
  }

  async handleSubmit(e){
    e.preventDefault();
    const { to, subject, message } = this.state
    const messageDetail = { reciever: to, subject, message}
    await this.props.sendMessage(messageDetail)
    await this.props.getMessages('');
    await this.props.getMessages('sent');
    await this.props.getMessages('unread');
    this.setState({
      success: this.props.messageSuccess,
      error: this.props.messageError || 'receiver is invalid'
    })
  }


  componentDidMount() {

    this.props.getMessages('');
    this.props.getMessages('sent');
    this.props.getMessages('unread');
    
  }
  sideToggle(e) {
    e.preventDefault()
   this.setState(prev => ({
    sidebarstate: !prev.sidebarstate
   }))
  }

  render() {
    const override = css`
      display: block;
      margin: 0 auto;
      border-color: red;
    `;

    const stylediv = {
      display: 'block'
    }

    return (
      <section className="main">
        <Header sideBar={e => this.sideToggle(e)}  name={this.props.name} dashstate={this.props.Dashboardstate} icon={this.state.sidebarstate ? <i className="fas fa-bars" /> : <i className="fas fa-times"></i>}/>
        <SideBar navname={this.state.sidebarstate ? 'nav' : 'nav_show'} click={(type) => this.handleModal('show')} />
        <Main />
        {this.props.loading ? (
          <div className="spinner_overlay">
            <div className="spinner-icon-container">
              <ScaleLoader css={override} sizeUnit={'px'} size={80} color={'#ffaf30'} loading={this.props.loading} />
            </div>
          </div>
        ) : (
          ''
        )}
        <div className={this.state.modal}>
          <form action="" className="message_form">
            <div className="message_container">
              <div className="message_wrapper">
                <ul>
                  <li className="message_wrapper_header">
                    <h1 className="message_header">New Message</h1>
                  </li>
                  <li className="message_wrapper_clear">
                    <div className="clear__overlay" onClick={(type) => this.handleModal('close')}>x</div>
                  </li>
                </ul>
              </div>
              <div className={this.state.success === 'message sent successfully' ? 'success' : 'error'} > {this.state.success || this.state.error} </div>

              <hr />

              <select className="" id="group__list" />

              <input 
              type="text" 
              placeholder="To" 
              name="to" 
              className="messageTo" 
              onChange={e => {this.handleInput(e)}}
              value={this.state.to} 
              />

              <input type="text" 
              placeholder="Subject" 
              name="subject" 
              className="messageSubject" 
              onChange={e => {this.handleInput(e)}}
              value={this.state.subject} 
              />

              <textarea
                name="message"
                id=""
                cols="30"
                rows="5"
                placeholder="Enter Your Message"
                className="messageContent"
                onChange={e => {this.handleInput(e)}}
                value={this.state.message} 
              />

              <hr />

              <button type="submit" className="send_message__btn" onClick = {e => {this.handleSubmit(e)}}>
               {this.props.sending ? 'sending...' : 'send'}
              </button>
              <button type="submit" className="draft_message__btn">
                Save
              </button>
            </div>
          </form>
        </div>
      </section>
    );
  }
}

const mapStateToProps = state => ({
  isAuthenticated: state.messages.isAuthenticated,
  inbox: state.messages.inbox,
  sent: state.messages.sent,
  unread: state.messages.unread,
  Dashboardstate: state.messages.dashboardstate,
  errors: state.messages.errors,
  loading: state.messages.isLoading,
  sending: state.messages.sending,
  messageSuccess: state.messages.messageSuccess,
  messageError: state.messages.messageError,
  name: state.auth.name
});
const mapDispatchToProps = {
  getMessages: type => messages(type),
  sendMessage: details => sendMessageAction(details),

};

export default connect(mapStateToProps, mapDispatchToProps)(withRouter(Dashboard))
