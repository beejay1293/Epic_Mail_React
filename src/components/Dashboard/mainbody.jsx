import React, { Component } from 'react';
import TopContent from './topcontent';
import Message from './messagebody';
import ViewSpecificMessage from './viewSpecificMessage';
import { connect } from 'react-redux';

class MainBody extends Component {
 render() {
   return (
  <div className="main__body">
     <TopContent />
     {this.props.body === 'viewMessages' ? <Message /> : <ViewSpecificMessage />}
  </div>
   )
 }
}
const mapStateToProps = state => ({
  isAuthenticated: state.messages.isAuthenticated,
  inbox: state.messages.inbox,
  sent: state.messages.sent,
  unread: state.messages.unread,
  Dashboardstate: state.messages.dashboardstate,
  errors: state.messages.errors,
  specificMessage: state.messages.specificMessage,
  body: state.messages.body
})
 

export default connect(mapStateToProps)(MainBody);
