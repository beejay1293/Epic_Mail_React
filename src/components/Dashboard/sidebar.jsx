import React from 'react';
import Compose from './compose';
import Btn from './btn';

const SideBar = ({click}) => (
  <div className="nav">
    <Compose click={click}/>
    <hr />
    <Btn />
  </div>
);

export default SideBar;
