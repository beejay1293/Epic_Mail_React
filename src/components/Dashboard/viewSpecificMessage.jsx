import React, { Component } from 'react';
import { connect } from 'react-redux'


class viewSpecificMessage extends Component {

    render() {
    return (
      <div className="read__email">
         <div className="subject__wrapper">
             <h1 className="subject">
            {this.props.specificMessage.subject}
          </h1> 
          <p className="time__created">{this.props.specificMessage.createdon}</p></div>  
          <div className="users">

            <h1 className="from"> {this.props.sender.firstname} {this.props.sender.lastname} </h1>
          <p className="epic">  { this.props.specificMessage.sender } </p>
            <p className="to">to: {this.props.specificMessage.receiver}</p>
          </div>      
           <div className="message__body">
           <p> {this.props.specificMessage.message}
            </p> 
            
          </div>
          <hr className="line"/>
        </div>

    )
}
}

const mapStateToProps = state => ({
    isAuthenticated: state.messages.isAuthenticated,
    inbox: state.messages.inbox,
    sent: state.messages.sent,
    unread: state.messages.unread,
    Dashboardstate: state.messages.dashboardstate,
    errors: state.messages.errors,
    specificMessage: state.messages.specificMessage,
    body: state.messages.body,
    sender: state.messages.sender,
    receiver: state.messages.receiver
})

export default connect(mapStateToProps)(viewSpecificMessage);
