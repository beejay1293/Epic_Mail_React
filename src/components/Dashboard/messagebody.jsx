import React, { Component } from 'react';
import SingleMessage from './SingleMessage';
import { messages, fetchSpecificMessage } from '../../actions'
import { connect } from 'react-redux';

export class MessageBody extends Component {
 state = {
   messages: []
 }
  render() {
    
    const styles = 'msg';
    if(this.props.Dashboardstate === 'Inbox'){
      this.state.messages = this.props.inbox 
    }else if(this.props.Dashboardstate === 'sent'){
     this.state.messages = this.props.sent
    } else if(this.props.Dashboardstate === 'unread'){
      this.state.messages = this.props.unread
    }
    return (
      <ul className="inbox__body">
        {this.state.messages.map(msg => (
          <SingleMessage
            style={styles}
            name= {msg.sender || msg.receiver}
            messages={msg.message}
            time={msg.createdon}
            key={msg.id}
            click={(id) => this.props.getMessage(msg.id)}
          />
        ))}

      </ul>
    );
  }
}
const mapDispatchToProps = {
    getMessage: (id) => fetchSpecificMessage(id)
}




const mapStateToProps = state => ({
  isAuthenticated: state.messages.isAuthenticated,
  inbox: state.messages.inbox,
  sent: state.messages.sent,
  unread: state.messages.unread,
  Dashboardstate: state.messages.dashboardstate,
  errors: state.messages.errors,
  specificMessage: state.messages.specificMessage,
  body: state.messages.body
})

export default connect(mapStateToProps, mapDispatchToProps)(MessageBody);

