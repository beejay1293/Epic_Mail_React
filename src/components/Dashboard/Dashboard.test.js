import React from 'react';
import { mount, shallow, render } from 'enzyme';
import Dashboard from './dashboard';
describe('Dashboard', () => {
  it('should render correctly in "debug" mode', () => {
    const DashboardPage = shallow(<Dashboard />);
  
    expect(DashboardPage).toMatchSnapshot();
  });
});

