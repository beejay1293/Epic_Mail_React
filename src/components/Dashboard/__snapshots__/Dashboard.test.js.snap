// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard should render correctly in "debug" mode 1`] = `
<ContextProvider
  value={null}
>
  <withRouter(Dashboard)
    Dashboardstate="Inbox"
    errors={Object {}}
    getMessages={[Function]}
    inbox={Array []}
    isAuthenticated={false}
    loading={false}
    messageError=""
    messageSuccess=""
    name=""
    sendMessage={[Function]}
    sending={false}
    sent={Array []}
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
    unread={Array []}
  />
</ContextProvider>
`;

exports[`test method test should test 1`] = `
<section
  className="main"
>
  <Header
    icon={
      <i
        className="fas fa-times"
      />
    }
    sideBar={[Function]}
  />
  <SideBar
    click={[Function]}
    navname="nav_show"
  />
  <Connect(MainBody) />
  <div
    className="close"
  >
    <form
      action=""
      className="message_form"
    >
      <div
        className="message_container"
      >
        <div
          className="message_wrapper"
        >
          <ul>
            <li
              className="message_wrapper_header"
            >
              <h1
                className="message_header"
              >
                New Message
              </h1>
            </li>
            <li
              className="message_wrapper_clear"
            >
              <div
                className="clear__overlay"
                onClick={[Function]}
              >
                x
              </div>
            </li>
          </ul>
        </div>
        <div
          className="error"
        >
           
           
        </div>
        <hr />
        <select
          className=""
          id="group__list"
        />
        <input
          className="messageTo"
          name="to"
          onChange={[Function]}
          placeholder="To"
          type="text"
          value=""
        />
        <input
          className="messageSubject"
          name="subject"
          onChange={[Function]}
          placeholder="Subject"
          type="text"
          value=""
        />
        <textarea
          className="messageContent"
          cols="30"
          id=""
          name="message"
          onChange={[Function]}
          placeholder="Enter Your Message"
          rows="5"
          value=""
        />
        <hr />
        <button
          className="send_message__btn"
          onClick={[Function]}
          type="submit"
        >
          send
        </button>
        <button
          className="draft_message__btn"
          type="submit"
        >
          Save
        </button>
      </div>
    </form>
  </div>
</section>
`;

exports[`test method test should test method 1`] = `
<section
  className="main"
>
  <Header
    icon={
      <i
        className="fas fa-bars"
      />
    }
    sideBar={[Function]}
  />
  <SideBar
    click={[Function]}
    navname="nav"
  />
  <Connect(MainBody) />
  <div
    className="close"
  >
    <form
      action=""
      className="message_form"
    >
      <div
        className="message_container"
      >
        <div
          className="message_wrapper"
        >
          <ul>
            <li
              className="message_wrapper_header"
            >
              <h1
                className="message_header"
              >
                New Message
              </h1>
            </li>
            <li
              className="message_wrapper_clear"
            >
              <div
                className="clear__overlay"
                onClick={[Function]}
              >
                x
              </div>
            </li>
          </ul>
        </div>
        <div
          className="error"
        >
           
           
        </div>
        <hr />
        <select
          className=""
          id="group__list"
        />
        <input
          className="messageTo"
          name="to"
          onChange={[Function]}
          placeholder="To"
          type="text"
          value=""
        />
        <input
          className="messageSubject"
          name="subject"
          onChange={[Function]}
          placeholder="Subject"
          type="text"
          value=""
        />
        <textarea
          className="messageContent"
          cols="30"
          id=""
          name="message"
          onChange={[Function]}
          placeholder="Enter Your Message"
          rows="5"
          value=""
        />
        <hr />
        <button
          className="send_message__btn"
          onClick={[Function]}
          type="submit"
        >
          send
        </button>
        <button
          className="draft_message__btn"
          type="submit"
        >
          Save
        </button>
      </div>
    </form>
  </div>
</section>
`;

exports[`test method test should test second input 1`] = `
<section
  className="main"
>
  <Header
    icon={
      <i
        className="fas fa-bars"
      />
    }
    sideBar={[Function]}
  />
  <SideBar
    click={[Function]}
    navname="nav"
  />
  <Connect(MainBody) />
  <div
    className="close"
  >
    <form
      action=""
      className="message_form"
    >
      <div
        className="message_container"
      >
        <div
          className="message_wrapper"
        >
          <ul>
            <li
              className="message_wrapper_header"
            >
              <h1
                className="message_header"
              >
                New Message
              </h1>
            </li>
            <li
              className="message_wrapper_clear"
            >
              <div
                className="clear__overlay"
                onClick={[Function]}
              >
                x
              </div>
            </li>
          </ul>
        </div>
        <div
          className="error"
        >
           
           
        </div>
        <hr />
        <select
          className=""
          id="group__list"
        />
        <input
          className="messageTo"
          name="to"
          onChange={[Function]}
          placeholder="To"
          type="text"
          value=""
        />
        <input
          className="messageSubject"
          name="subject"
          onChange={[Function]}
          placeholder="Subject"
          type="text"
          value=""
        />
        <textarea
          className="messageContent"
          cols="30"
          id=""
          name="message"
          onChange={[Function]}
          placeholder="Enter Your Message"
          rows="5"
          value=""
        />
        <hr />
        <button
          className="send_message__btn"
          onClick={[Function]}
          type="submit"
        >
          send
        </button>
        <button
          className="draft_message__btn"
          type="submit"
        >
          Save
        </button>
      </div>
    </form>
  </div>
</section>
`;

exports[`test method test should test third input 1`] = `
<section
  className="main"
>
  <Header
    icon={
      <i
        className="fas fa-bars"
      />
    }
    sideBar={[Function]}
  />
  <SideBar
    click={[Function]}
    navname="nav"
  />
  <Connect(MainBody) />
  <div
    className="close"
  >
    <form
      action=""
      className="message_form"
    >
      <div
        className="message_container"
      >
        <div
          className="message_wrapper"
        >
          <ul>
            <li
              className="message_wrapper_header"
            >
              <h1
                className="message_header"
              >
                New Message
              </h1>
            </li>
            <li
              className="message_wrapper_clear"
            >
              <div
                className="clear__overlay"
                onClick={[Function]}
              >
                x
              </div>
            </li>
          </ul>
        </div>
        <div
          className="error"
        >
           
           
        </div>
        <hr />
        <select
          className=""
          id="group__list"
        />
        <input
          className="messageTo"
          name="to"
          onChange={[Function]}
          placeholder="To"
          type="text"
          value=""
        />
        <input
          className="messageSubject"
          name="subject"
          onChange={[Function]}
          placeholder="Subject"
          type="text"
          value=""
        />
        <textarea
          className="messageContent"
          cols="30"
          id=""
          name="message"
          onChange={[Function]}
          placeholder="Enter Your Message"
          rows="5"
          value=""
        />
        <hr />
        <button
          className="send_message__btn"
          onClick={[Function]}
          type="submit"
        >
          send
        </button>
        <button
          className="draft_message__btn"
          type="submit"
        >
          Save
        </button>
      </div>
    </form>
  </div>
</section>
`;
